src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -126,18 +134,29 @@ internal static FormulaValue FromJson(JsonElement element, FormulaValueJsonSeria
                             dt2 = dt2.ToUniversalTime();
                         }
 
-                        if (dt2.Kind == DateTimeKind.Unspecified)
+                        if (settings.ResultTimeZone == TimeZoneInfo.Utc && dt2.Kind == DateTimeKind.Unspecified)
                         {
                             dt2 = new DateTime(dt2.Ticks, DateTimeKind.Utc);
                         }
 
-                        return DateTimeValue.New(dt2);
+                        return DateTimeValue.New(DateTimeValue.GetConvertedDateTimeValue(dt2, settings.ResultTimeZone));
                     }
                     else if (formulaType is DateTimeNoTimeZoneType)
                     {
                         DateTime dt3 = element.GetDateTime();
                         return DateTimeValue.New(TimeZoneInfo.ConvertTimeToUtc(dt3));
                     }
+                    else if (formulaType is TimeType)
+                    {
+                        var timeString = element.GetString();
+                        if (TimeSpan.TryParseExact(timeString, @"hh\:mm\:ss\.FFFFFFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out var res) ||
```

Comment:
>FFFFFFF [](http://example.com/codeflow?start=77&length=7)

Does it need to be 0 or 7 decimal places? If we have something like 12:34:56.789, should it work?

Reply:
using the same piece from TimeValue Function for it to be consistent